src/io/parquet/read/deserialize/binary/utils.rs

```diff
@@ -56,13 +56,21 @@ impl<O: Offset> Binary<O> {
         offsets.push(O::default());
         Self {
             offsets: Offsets(offsets),
-            values: Vec::with_capacity(capacity * 24),
+            values: Vec::with_capacity(capacity.min(100) * 24),
             last_offset: O::default(),
         }
     }
 
     #[inline]
     pub fn push(&mut self, v: &[u8]) {
+        if self.offsets.0.len() == 101 && self.offsets.0.capacity() > 101 {
+            let bytes_per_row = self.values.len() / 100 + 1;
```

Comment:
Good idea on the sampling 100 items first. :+1:

Reply:
<img width="852" alt="image" src="https://user-images.githubusercontent.com/3325189/204945173-f1020ffc-51b0-43ae-8fac-e2ef7dd9f104.png">

Idea from https://www.infoq.cn/article/xwijc1afv3cn5znj2sdi (it's in Chinese)